AI/BattleAI/AttackPossibility.cpp

```diff
@@ -117,8 +117,8 @@ float AttackPossibility::attackValue() const
 
 float hpFunction(uint64_t unitHealthStart, uint64_t unitHealthEnd, uint64_t maxHealth)
 {
-	float ratioStart = static_cast<float>(unitHealthStart) / maxHealth;
-	float ratioEnd = static_cast<float>(unitHealthEnd) / maxHealth;
+	auto ratioStart = static_cast<float>(unitHealthStart) / maxHealth;
+	auto ratioEnd = static_cast<float>(unitHealthEnd) / maxHealth;
```

Comment:
To be honest in case of built-in types explicit type name might be better.

Not against this change though.

Reply:
Probably OK as it. Unless somebody else prefers old version.
At the very least it should reduce amount of diagnostics at sonarcloud, potentially allowing to find actual issues.

Reply:
At my company we vise versa prefer exact types instead of auto for better readability. In C++ there are cases when typenames are very big and auto is simpler to write. This is why I believe we exist in such indetermined state regarding this keyword and everyone makes it in a way they prefer. In this exact case I do not see a big difference and have no exact preference so.